### ISSUE TITLE
keeps deleting existing code even when not required and specifically asked not to.

### ISSUE THREAD

**Issue by IrfanKheiri** [cline/cline#596]
### Which API Provider are you using?

Anthropic

### Which Model are you using?

claude sonnet 3.5 20241022

### What happened?

keeps deleting existing code even when not required and specifically asked not to.
output:
I apologize for my previous attempt. You're right - I should only modify the main function while preserving all other code. Let me read the file again and then make targeted changes only to the main function.
still marked most of the code for deletion.
rejecting causes it to start again with same behaviour.

### Steps to reproduce

1. existing code with class having multiple methods (Python)
2. prompt to add code in main function to test each method in class
3. marks almost the entire existing code for deletion.
4. press reject on access request.
5. goes and again marks most of the code for deletion.

**Comment by martincrumlish** [comment]
I noticed this issue today also, after the Sonnet update possibly.

Yesterday, while using Cline all day, it did not delete code and replace it with comments like "//existing code goes here". The recent update that amended the system prompt seemed to resolve this issue, with the previous sonnet model.

**Comment by tluyben** [comment]
Yeah, this is now absolutely rampant to the point i just added haiku to check the result and then say; "it seems you removed again a lot of functioanlity that is vital, please don't do that" and then it gets fixed.

**Comment by tluyben** [comment]
Is this related https://www.reddit.com/r/ClaudeAI/comments/1g8jtfm/i_think_this_is_like_really_good_update/ and how do we switch it off in the API?

**Comment by ccrvlh** [comment]
Related to #583

**Comment by DaDaDarryl** [comment]
Experiencing the same on Anthropic Claude Sonnet 3.5 20241022. As well as the obsession to remove all comments. I'm now asking Cline to only provide the list of changes then I manually add them in my files - sometimes it listens, other times it still edits the file, and erroneously removes code.

**Comment by Kaddtechnologies** [comment]
this needs to be priority 1 on the list to fix

**Comment by tluyben** [comment]
The removals of comments I didn't have before, but now I do, often it says 'let fix this file ' and then it ONLY removes the comments, nothing else...

**Comment by celestial-vault** [comment]
This is a model issue and can be improved with prompting
